### ISSUE TITLE
unable to get why negative value of class_loss is taken

### ISSUE THREAD

**Issue by surekhag28** [utkuozbulak/pytorch-cnn-visualizations#71]
I am quite unable to understand why we have taken negative value of class_loss? As far as I understand main intention should be to increase the score of the particular class for which we are trying to generate the image. Also why you are not taking softmax before doing backpropagation on the target loss? Please correct me if I am wrong as I am confused with the approach.

**Comment by utkuozbulak** [comment]
Negative loss is usually taken to maximize an output. You can write down derivation yourself and see what maximizes and what minimizes.

Softmax is not preferred because it limits the signal.
